minecraft/src/main/java/net/fabricmc/loader/impl/game/minecraft/MinecraftGameProvider.java

```diff
@@ -207,15 +232,15 @@ public boolean locateGame(FabricLauncher launcher, String[] args, ClassLoader cl
 		return true;
 	}
 
-	private boolean processBundlerJar(ClassLoader cl) {
+	private boolean processBundlerJar(Path path) {
 		if (envType != EnvType.SERVER) return false;
 
 		// determine urls by running the bundler and extracting them from the context class loader
 
 		URL[] urls;
 
 		try {
-			ClassLoader bundlerCl = new ClassLoader(cl) {
+			ClassLoader bundlerCl = new URLClassLoader(new URL[] { path.toUri().toURL() }, MinecraftGameProvider.class.getClassLoader()) {
```

Comment:
Might as well put this as a try with resources if it's only a temporary `URLCLassLoader`